libraries/AP_Scripting/AP_Scripting.h

```diff
@@ -44,9 +44,16 @@ class AP_Scripting
    // User parameters for inputs into scripts 
    AP_Float _user[4]; 
 
+    enum class relp_type {
+        REPL_FILE,
+        REPL_SERIAL,
+    };
+
     struct terminal_s {
         int output_fd;
         off_t input_offset;
+        AP_HAL::UARTDriver *uart; 
```

Comment:
Sorry to come back to this late but can this be rearranged a bit? I think I'd prefer to see a pair of structs that contain the relevant info for the UART vs the file one seperately, and then see both of them inside this struct, or a union of them here. My concern is it seems like it's increasingly easy to use the members for different things.